flow/project.py

```diff
@@ -291,13 +304,19 @@ def __init__(self, id, name, job, cmd, directives=None):
         self.directives._keys_set_by_user = keys_set_by_user
 
     def __str__(self):
-        return "{}({})".format(self.name, self.job)
+        if len(self._jobs) == 1:
+            return "{}({})".format(self.name, str(self._jobs[0]))
+        else:
```

Comment:
I keep repeating that I'd like to treat all operations as similar as possible and that the "classic single-argument" operations are simply aggregate operations of size 1. This should be reflected here as well and I would very much prefer to use the same representation of the class for size 1 and larger. Having a fundamentally different representation for size 1 is unexpected and will likely cause a lot of confusion.

This is not a trivial issue, ensuring that users (and developers!) understand that there is nothing fundamentally different about an aggregate operation of size 1 is crucial for a good understanding of the code logic and ultimately usage.

Reply:
That's a nice point. 
Thanks for this!

Reply:
This is not a really hard decision to make, but can you please suggest me whether I should, for now, not check the length of jobs as I'm sure that it'll always be one?

Reply:
Add `assert len(self._jobs) == 1` here and then remove it once you need to drop that assertion in a different change set.